src/zope/interface/registry.py

```diff
@@ -184,6 +145,9 @@ def __init__(self, name='', bases=()):
         self._init_registrations()
         self.__bases__ = tuple(bases)
 
+        if hasattr(self, '_v_utility_registrations_cache'):
+            del self._v_utility_registrations_cache
```

Comment:
I added a comment to address this.  It would be good for someone to make the testing interactions across the packages cleaner. That is out of scope of this change.

With the comment, at least there's now something describing the current dual responsibilities of __init__.